includes/class-wc-payments-order-success-page.php

```diff
@@ -14,11 +14,33 @@ class WC_Payments_Order_Success_Page {
 	 * Constructor.
 	 */
 	public function __construct() {
+		add_filter( 'woocommerce_thankyou_order_id', [ $this, 'update_order_meta' ] );
 		add_filter( 'woocommerce_thankyou_order_received_text', [ $this, 'show_woopay_thankyou_notice' ], 10, 2 );
 		add_action( 'woocommerce_thankyou_woocommerce_payments', [ $this, 'show_woopay_payment_method_name' ] );
 		add_action( 'wp_enqueue_scripts', [ $this, 'enqueue_scripts' ] );
 	}
 
+	public function update_order_meta( $order_id ) {
```

Comment:
The filter expects an ID to be returned, specifically the ID for the Order that will be displayed on the Order Received page.